pallets/gear/src/manager/journal.rs

```diff
@@ -176,10 +176,18 @@ where
             let _ = TaskPoolOf::<T>::delete(bn, ScheduledTask::PauseProgram(id_exited));
 
             match p {
-                Program::Active(program) => Self::remove_gas_reservation_map(
-                    id_exited,
-                    core::mem::take(&mut program.gas_reservation_map),
-                ),
+                Program::Active(program) => {
+                    Self::remove_gas_reservation_map(
+                        id_exited,
+                        core::mem::take(&mut program.gas_reservation_map),
+                    );
+
+                    Self::clean_inactive_program(
```

Comment:
I guess all the funds will NOT be returned to user if program is NOT active?

Reply:
Sorry, didn't catch what you meant. `clean_inactive_program` transfers all free funds of the program to its so called inheritor. It is its main function actually